src/adagrams.js

```diff
@@ -1,15 +1,115 @@
+const LETTERPOOL = {
+  'A': 9, 
+  'B': 2, 
+  'C': 2, 
+  'D': 4, 
+  'E': 12, 
+  'F': 2, 
+  'G': 3, 
+  'H': 2, 
+  'I': 9, 
+  'J': 1, 
+  'K': 1, 
+  'L': 4, 
+  'M': 2, 
+  'N': 6, 
+  'O': 8, 
+  'P': 2, 
+  'Q': 1, 
+  'R': 6, 
+  'S': 4, 
+  'T': 6, 
+  'U': 4, 
+  'V': 2, 
+  'W': 2, 
+  'X': 1, 
+  'Y': 2, 
+  'Z': 1
+}
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+    let letterFreq = {};
+    let selectedLetter = [];
+    while ((selectedLetter.length) < 10) {
+      let random = Object.keys(LETTERPOOL)[Math.floor(Math.random()*Object.keys(LETTERPOOL).length)];
+          if (random in letterFreq) {
+              letterFreq[random] += 1;
+          } else {
+              letterFreq[random] = 1;
+          if (letterFreq[random] <= LETTERPOOL[random]){
+            selectedLetter.push(random)
+          }        
+          }        
+    }
+    return (selectedLetter)
 };
```

Comment:
Overall I see what you were trying to achieve here, just a few small adjustments!

 1) Your function currently forces a false randomness. It will randomly pick one of the 26 letters and remove one, 
 but it ignores the fact that we have different letters with different frequencies. To get around this, the main 
 suggestion would be to make a global dictionary like the one you had, and then write a helper function that 
 converts that dictionary into a list or string that holds every single letter possible. Then we can use that helper 
 function to create our collection of letters as opposed to making a deep copy! This will also make it so that you don't have tp build a frequency map to make those comparisons!

 2) If we have that collection of all the letters, it makes it really easy to remove letters as you use them!